packages/scalecube-microservice/src/MicroService/ServiceRegistry.ts

```diff
@@ -0,0 +1,61 @@
+import { ServiceEndPoint } from '../api/Service';
+import { generateIdentifier, isValidRawService } from '../helpers/utils';
+import { getServiceName } from '../helpers/utils';
+
+export const lookUp = ({ serviceRegistry, methodName }) => serviceRegistry[methodName];
+
+export const updateServiceRegistry = ({ serviceRegistry, rawService }) => {
+  if (isValidRawService(rawService)) {
+    servicesFromRawService(rawService).forEach((service) => addServiceToRegistry({ serviceRegistry, service }));
+  }
+
+  return { ...serviceRegistry };
+};
+
+const addServiceToRegistry = ({ serviceRegistry, service }) => {
+  const nameSpace = `${getServiceName(service)} ${service.methodName}`;
+
+  serviceRegistry[nameSpace] = {
```

Comment:
I would suggest not to modify serviceRegistry directly, but to return immutable copy to achieve putiry

Reply:
export const addServiceToRegistry = ({ serviceRegistry, service }) => {
  const nameSpace = `${getServiceName(service)} ${service.methodName}`;

  serviceRegistry[nameSpace] = {
    ...(serviceRegistry[nameSpace] || {}),
    ...serviceEndPoint(service),
  };

  return { ...serviceRegistry };
};

we return the immutable copy of the serviceRegistry